File: Python/70_Climbing_Stairs.py
import random


class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        a = b = 1
        for i in range(2, n + 1):
            a, b = b, a + b
        return b


def test():
    s = Solution()
    stair_num = random.randint(1, 20)
    print "Number of layers: " + str(stair_num) + "\tNumber of steps: " + str(s.climbStairs(stair_num))


if __name__ == '__main__':
    test()

File: C#/70_Climbing_Stairs.cs
using System;

namespace LeetCode
{
    public class Solution
    {
        public int ClimbStairs(int n)
        {
            int a = 1;
            int b = 1;
            for (int i = 2; i < n + 1; i++)
            {
                int temp = a + b;
                a = b;
                b = temp;
            }
            return b;
        }

        //Testcase
        static void Main(string[] args)
        {
            Solution s = new Solution();
            int stair_num = new Random().Next(1, 21);
            Console.Write("Number of layers: " + stair_num +
                "\tNumber of steps: " + s.ClimbStairs(stair_num));           
        }
    }
}